ibis/expr/api.py

```diff
@@ -1904,7 +1897,11 @@ def _timestamp_time(arg):
 
 _join_classes = {
     'inner': _ops.InnerJoin,
+    'any_inner': _ops.AnyInnerJoin,
+    'all_inner': _ops.AllInnerJoin,
     'left': _ops.LeftJoin,
+    'any_left': _ops.AnyLeftJoin,
+    'all_left': _ops.AllLeftJoin,
```

Comment:
After reading the clickhouse docs [here](https://clickhouse.yandex/docs/en/query_language/queries.html#join-clause) I don't think we need most of these additional APIs:

1. `all_inner` is `inner_join`
1. `any_left` is `left_semi_join`/`semi_join`
1. `all_left` is `left_join`

The only one that appears to be unique is `any_inner`. I think we should call this `inner_semi` to be consistent with `left_semi`.

The clickhouse translator (specifically the `TableSetFormatter`) can deal with placing the correct `ANY`/`ALL` keyword depending on the type of join being translated.

Reply:
Ok, I'll use those. How about array join?

Reply:
We don't have an expression for that yet. I've created #1146 to track it.